iati_datastore/iatilib/crawler.py

```diff
@@ -166,30 +113,17 @@ def fetch_resource(resource):
     :param resource:
     :return:
     '''
-    headers = {
-        'User-Agent': 'codeforIATI datastore classic',
-    }
-    if resource.last_succ:
-        headers['If-Modified-Since'] = http_date(resource.last_succ)
-    if resource.etag:
-        headers["If-None-Match"] = resource.etag
-    resp = requests.get(resource.url, headers=headers)
-    resource.last_status_code = resp.status_code
-    resource.last_fetch = datetime.datetime.utcnow()
-    if resp.status_code == 200:
-        if isinstance(resp.content, bytes):
-            resource.document = resp.content
-        else:
-            resource.document = resp.content.encode()
-        if "etag" in resp.headers:
-            resource.etag = resp.headers.get('etag')
-        else:
-            resource.etag = None
-        resource.last_succ = datetime.datetime.utcnow()
-        resource.last_parsed = None
-        resource.last_parse_error = None
-    if resp.status_code == 304:
-        resource.last_succ = datetime.datetime.utcnow()
+    dataset = Dataset.query.get(resource.dataset_id)
+    fname = '__iatikitcache__/registry/data/{0}/{1}.xml'.format(
+        dataset.publisher, dataset.name)
+    with open(fname, 'rb') as f:
+        content = f.read()
+
+    resource.document = content
+    resource.last_succ = iatikit.data().last_updated
+    resource.last_parsed = None
+    resource.last_parse_error = None
```

Comment:
Yeah… I think this is a bit weird but okay. I guess the idea is that this dataset has been modified since it was last seen, so we haven’t yet parsed this newly modified version. So this serves as a flag that the dataset needs reparsing.

An alternative approach would be to not set these to `None`, and instead determine if the dataset needs reparsing by checking if `last_succ > last_parsed`.